genie-agent/src/main/java/com/netflix/genie/agent/cli/ArgumentConverters.java

```diff
@@ -175,4 +176,54 @@ public JsonNode convert(final String value) {
             }
         }
     }
+
+    static final class UriOrLocalPathConverter implements IStringConverter<String> {
+        static final String ATTACHMENT_HELP_MESSAGE = "ATTACHMENTS:\n"
+            + "Different kinds of job-specific files can be attached to a job.\n"
```

Comment:
This is likely a minor thing but I worry about causing confusion with users by overloading the term "attachment" in different contexts. Historically, and I'm not saying we can't improve or correct this if we think it's wrong, we've used `attachment` to mean a file not available anywhere else that is being sent along with the job request to Genie. 

The wording in this help message makes it seem like all different types of files are these kinds of historical `attachments`. We've used `dependency` to mean something available somewhere else. 

I'm not sure this needs to be solved right now I just want to avoid users getting confused when we're debugging something and they say it's an `attachment`. If that makes sense.

This all may be part of a bigger discussion about how we evolve file handling for jobs but I wanted to make sure I pointed it out so at least we think through it. This is likely ok for now as most end users won't see it since internally we wrap in that Python lib.

Reply:
I agree and this is actually my attempt at setting things straight by moving our lexicon in a specific direction.
In V3, we have these 3 types of files, plus we have a fourth: "anonymous" attachments (anonymous because they are not recorded in the request or anywhere else). 

I think we should move away from anonymous attachments and instead force the users to think in terms of configs, and dependencies and setup. 
And consider the word "attachment" for a config or a dependency or a setup file that is uploaded along with the job request.